implementation/src/main/java/io/smallrye/config/EnvConfigSource.java

```diff
@@ -31,9 +31,11 @@
  */
 public class EnvConfigSource extends AbstractConfigSource {
     private static final long serialVersionUID = -4525015934376795496L;
+
     private static final int DEFAULT_ORDINAL = 300;
+    private static final String NULL_VALUE = "io.smallrye.config.EnvConfigSource.nullValue";
```

Comment:
Rather than rely on a unique string value - which could (at least theoretically) be a legitimate value - what about using an object with unique identity instead?

```suggestion
    private static final String NULL_VALUE = new String("");
```

Then compare by identity:
```java
if (cachedValue == NULL_VALUE) {
```

Alternatively, a unique object of type `Object` may be used (this works best if the value type of the `Map` be `Object`, which would require that the result of the `get` method must be cast on return - but this cast happens internally anyway so it doesn't impact performance or anything of that sort).

Reply:
I actually had that code before, but it fails if you serialize the source. 

I didn't want to make the cache transient, but maybe we need to do it. What do you think?

Reply:
Well probably a Serializable object should work.

Reply:
For serialization we probably want a different strategy.  Serializing the cache from one system and deserializing it on another doesn't make any sense; we probably want replacement objects (again).  I'll add a suggestion down below.